src/analysis/construct.ml

```diff
@@ -62,8 +62,10 @@ module Util = struct
     in
     tbl
 
-  let prefix env ~env_check path name =
-    to_shortest_lid ~env ~env_check ~name path
+  let prefix env ~env_check ~omit_if_possible path name =
+    match omit_if_possible, Warnings.is_active (Disambiguated_name "") with
+    | true, false -> Longident.Lident name
+    | _ -> to_shortest_lid ~env ~env_check ~name path
```

Comment:
Done.  My thinking was that it would be nice to force callsites to decide whether they might benefit from omitting the prefix, but I agree it is inelegant.